src/zope/paste/tests.py

```diff
@@ -124,14 +124,27 @@ def test_basic_serve():
 def test_serving_test_app():
     """Test serving a real app.
 
+    Setup logging, so we get the message ``Serving on ...`` into our doctest
+    output:
+
+    >>> import logging
+    >>> logger = logging.getLogger('waitress')
+    >>> logger.addHandler(logging.StreamHandler(sys.stdout))
+    >>> logger.setLevel(logging.INFO)
```

Comment:
I'd rather do that in `setUp()`.

(And remove the handler in `tearDown()`, if we ever have two tests relying on waitress in some way, it won't be fun to have global state mutations persist from one test to another.)

Reply:
Good idea.

Reply:
> Good idea.

It seemed good, but it is too early. `doctest` changes `sys.stdout` to an internal stream. But this seems to be done after `setUp` is called. At least: it does not work when setting login up this way in `setUp`. So I added a teardown to the doctest, see https://github.com/zopefoundation/zope.paste/pull/3/commits/3d1e0ea9faa3ce8744e688ea1fbd2969edb0fe34.